ocaml/stdlib/float_u.mli

```diff
@@ -0,0 +1,344 @@
+   Array submodule, which we don't have here, so I haven't done something
+   similar.  If we have an Array submodule here in the future, reconsider. *)
+
+(** Unboxed floating-point arithmetic.  This file primarily duplicates
+    functionality from the [Float] module, but for [float#].
+
+    OCaml's floating-point numbers follow the
+    IEEE 754 standard, using double precision (64 bits) numbers.
+    Floating-point operations never raise an exception on overflow,
+    underflow, division by zero, etc.  Instead, special IEEE numbers
+    are returned as appropriate, such as [infinity] for [1.0 /. 0.0],
+    [neg_infinity] for [-1.0 /. 0.0], and [nan] ('not a number')
+    for [0.0 /. 0.0].  These special numbers then propagate through
+    floating-point computations as expected: for instance,
+    [1.0 /. infinity] is [0.0], basic arithmetic operations
+    ([+.], [-.], [*.], [/.]) with [nan] as an argument return [nan], ...
+*)
+
+(* CR layouts v5: add back all the constants in this module (e.g., [zero] and
+   [infinity]) when we we support [float64]s in structures. *)
+
+(* Unboxed-specific stuff at the top. *)
+external to_float : float# -> (float[@local_opt]) = "%box_float"
+(** Box a [float#] *)
+
+external of_float : (float[@local_opt]) -> float# = "%unbox_float"
+(** Unbox a boxed [float] *)
+
+(* Below here, everything also appears in [Float], though most things are
+   externals in that module. *)
+
+val neg : float# -> float#
+(** Unary negation. *)
+
+val add : float# -> float# -> float#
+(** Floating-point addition. *)
+
+val sub : float# -> float# -> float#
+(** Floating-point subtraction. *)
+
+val mul : float# -> float# -> float#
+(** Floating-point multiplication. *)
+
+val div : float# -> float# -> float#
+(** Floating-point division. *)
+
+val fma : float# -> float# -> float# -> float#
+(** [fma x y z] returns [x * y + z], with a best effort for computing
+   this expression with a single rounding, using either hardware
+   instructions (providing full IEEE compliance) or a software
+   emulation.
+
+   On 64-bit Cygwin, 64-bit mingw-w64 and MSVC 2017 and earlier, this function
+   may be emulated owing to known bugs on limitations on these platforms.
+   Note: since software emulation of the fma is costly, make sure that you are
+   using hardware fma support if performance matters. *)
+
+val rem : float# -> float# -> float#
+(** [rem a b] returns the remainder of [a] with respect to [b].  The returned
+    value is [a -. n *. b], where [n] is the quotient [a /. b] rounded towards
+    zero to an integer. *)
+
+val succ : float# -> float#
+(** [succ x] returns the floating point number right after [x] i.e.,
+   the smallest floating-point number greater than [x].  See also
+   {!next_after}. *)
+
+val pred : float# -> float#
+(** [pred x] returns the floating-point number right before [x] i.e.,
+   the greatest floating-point number smaller than [x].  See also
+   {!next_after}. *)
+
```

Comment:
No need to fix this in retrospect, but I think it would be a bit better to keep the unused definitions from float.mli here, but commented out. This would make auditing completion easier and would also make the v5 work easier. (Fixing this now would basically be doing that work, so no need.)